java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitutionSolver.java

```diff
@@ -200,16 +205,21 @@ private TypeSubstitution inferTypeSubstitution(JavaSymbol.MethodJavaSymbol metho
     return substitution;
   }
 
-  private TypeSubstitution inferTypeSubstitution(JavaSymbol.MethodJavaSymbol method, TypeSubstitution currentSubstitution, boolean isLastParam, JavaType formalType,
-    JavaType argType) {
+  private TypeSubstitution inferTypeSubstitution(JavaSymbol.MethodJavaSymbol method, TypeSubstitution currentSubstitution, JavaType formalType, JavaType argType,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 7, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1142)